c/tskit/tables.c

```diff
@@ -4518,31 +4481,86 @@ table_sorter_sort_mutations(table_sorter_t *self)
     return ret;
 }
 
-static int
-table_sorter_run(table_sorter_t *self, tsk_size_t edge_start)
+int
+tsk_table_sorter_run(tsk_table_sorter_t *self, tsk_bookmark_t *start)
 {
     int ret = 0;
+    tsk_size_t edge_start = 0;
 
-    ret = table_sorter_sort_edges(self, edge_start);
+    if (start != NULL) {
+        if (start->edges > self->tables->edges.num_rows) {
+            ret = TSK_ERR_EDGE_OUT_OF_BOUNDS;
+            goto out;
+        }
+        edge_start = start->edges;
+
+        /* For now, if migrations, sites or mutations are non-zero we get an error.
+         * This should be fixed: https://github.com/tskit-dev/tskit/issues/101
+         */
+        if (start->migrations != 0 || start->sites != 0 || start->mutations != 0) {
+            ret = TSK_ERR_SORT_OFFSET_NOT_SUPPORTED;
+            goto out;
+        }
+    }
+    /* The indexes will be invalidated, so drop them */
+    ret = tsk_table_collection_drop_index(self->tables, 0);
     if (ret != 0) {
         goto out;
     }
-    ret = table_sorter_sort_sites(self);
+    if (self->sort_edges != NULL) {
```

Comment:
Is there a utility to not sorting the edges, for some case?  I.e., is that why passing NULL produces that behavior, instead of just asserting that self->sort_edges must be non-NULL?

Reply:
You can imagine the following type of logic (psueodcode) being useful for very large tables:

```cpp
std::packaged_task<int()> efficient_edge_sorter(...);
std::future<int> sorter_future = efficient_edge_sorter.get_future();
tsk_table_collection_sort(...); // sort_edges = nullptr
sorter_future.wait();
```

The result is that the edge sorting is done asynchronously to the sorting of the other tables, which only depend on times, meaning the contents of the node table.